azure/scope/machinepoolmachine.go

```diff
@@ -149,6 +152,24 @@ func NewMachinePoolMachineScope(params MachinePoolMachineScopeParams) (*MachineP
 	}, nil
 }
 
+// ScaleSetVMSpec returns the VMSS VM spec.
+func (s *MachinePoolMachineScope) ScaleSetVMSpec() azure.ResourceSpecGetter {
```

Comment:
Should we add a unit test for this and `SetVMSSVMState` below? They're far from complicated but it seems they don't have coverage.